#### Lab Assignment 1 - Jack Travers
#### Student ID: 200013389
#### GitHub URL: https://github.com/Jacktravers81/PY4SA23_Assignment

##### Section One: Python Basics

Task 1: Create an If...Else statement that will test whether a number is divisible by three.

In [79]:
def divisible (n):
    if n % 3 == 0:
        print (n, "is divisible by 3")
    else:
        print (n, "is not divisible by 3")
divisible (11)

11 is not divisible by 3


Task 2: Create an If...Else statement that will test whether a type of fruit, represented as a text string, is in a list of acceptable fruits (apple, orange, pear, kiwi, or strawberry).

In [43]:
fruit = ["apple", "orange", "pear", "kiwi", "strawberry"]

def library(check):
    if check in fruit:
        print (check, "is acceptable")
    else:
        print (check, "is not acceptable")

library("berry")

berry is not acceptable


Task 3: Create a function to calculate the distance between two coordinates using the haversine
formula. Write the following formula where the input parameters are a pair of coordinates as two
lists.

In [44]:
# COME BACK TO ME!!!!! 

a = sin²(φB - φA/2) + cos φA * cos φB * sin²(λB - λA/2)
c = 2 * atan2( √a, √(1−a) )
d = R ⋅ c

SyntaxError: invalid character '²' (U+00B2) (1688034982.py, line 5)

##### Section Two: Pandas and NumPy

Question 1: How many trees are of the Quercus or Acer genus?

In [80]:
import numpy as np 
import pandas as pd 
trees_df = pd.read_csv("Assignment 2 Data\portland_park_trees.csv", index_col=30)

trees_df.value_counts("Genus")

Genus
Pseudotsuga      6783
Acer             3586
Quercus          2089
Prunus           1333
Pinus             983
                 ... 
Eucommia            1
Eriobotrya          1
Daphniphyllum       1
Schima              1
Illicium            1
Length: 117, dtype: int64

Answers:
Genus *Acer*: 3586,
Genus *Quercus*: 2089

Question 2: : How many trees are of the Quercus or Acer genus and have a DBH larger than 50 inches?

In [81]:
import numpy as np 
import pandas as pd 
trees_df = pd.read_csv("Assignment 2 Data\portland_park_trees.csv", index_col=30)

trees_df.value_counts("Genus")
"DBH" = > 50

Genus
Pseudotsuga      0.265646
Acer             0.140440
Quercus          0.081812
Prunus           0.052205
Pinus            0.038498
                   ...   
Eucommia         0.000039
Eriobotrya       0.000039
Daphniphyllum    0.000039
Schima           0.000039
Illicium         0.000039
Length: 117, dtype: float64